File: frontend/src/services/appointment.service.js
import api from './api';

export const getAllAppointments = async () => {
  const response = await api.get('/appointment/all');
  return response.data;
};

export const getDoctorAppointments = async (doctorName) => {
  const response = await api.get(`/appointment/doctor/${doctorName}`);
  return response.data;
};

export const bookAppointment = async (appointmentData) => {
  // appointmentData: { patientName, doctorName, appointmentDate }
  const response = await api.post('/appointment/book', appointmentData);
  return response.data;
};

export const updateAppointment = async (id, appointmentData) => {
  const response = await api.put(`/appointment/update/${id}`, appointmentData);
  return response.data;
};

export const deleteAppointment = async (id) => {
  const response = await api.delete(`/appointment/delete/${id}`);
  return response.data;
};

File: ers-backend/src/main/java/com/ers/ersbackend/service/AppointmentService.java
package com.ers.ersbackend.service;

import com.ers.ersbackend.model.Appointment;
import com.ers.ersbackend.repository.AppointmentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class AppointmentService {

    @Autowired
    private AppointmentRepository appointmentRepository;

    // ✅ CREATE
    public Appointment bookAppointment(Appointment appointment) {
        return appointmentRepository.save(appointment);
    }

    // ✅ READ ALL
    public List<Appointment> getAllAppointments() {
        return appointmentRepository.findAll();
    }

    // ✅ READ BY DOCTOR
    public List<Appointment> getAppointmentsByDoctor(String doctorName) {
        return appointmentRepository.findByDoctorName(doctorName);
    }

    // ✅ READ ONE
    public Appointment getAppointmentById(Long id) {
        return appointmentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Appointment not found with id: " + id));
    }

    // ✅ UPDATE
    public Appointment updateAppointment(Long id, Appointment updatedAppointment) {
        Appointment appointment = getAppointmentById(id);

        appointment.setPatientName(updatedAppointment.getPatientName());
        appointment.setDoctorName(updatedAppointment.getDoctorName());
        appointment.setAppointmentDate(updatedAppointment.getAppointmentDate());

        return appointmentRepository.save(appointment);
    }

    // ✅ DELETE
    public void deleteAppointment(Long id) {
        Appointment appointment = getAppointmentById(id);
        appointmentRepository.delete(appointment);
    }
}